x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvMedianAbsoluteDeviation.java

```diff
@@ -0,0 +1,353 @@
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.compute.ann.MvEvaluator;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.IntBlock;
+import org.elasticsearch.compute.data.LongBlock;
+import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
+import org.elasticsearch.xpack.esql.EsqlIllegalArgumentException;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.core.util.NumericUtils;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.planner.PlannerUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isType;
+import static org.elasticsearch.xpack.esql.core.type.DataType.isRepresentable;
+import static org.elasticsearch.xpack.esql.core.util.NumericUtils.unsignedLongSubtractExact;
+
+/**
+ * Reduce a multivalued field to a single valued field containing the median absolute deviation of the values.
+ */
+public class MvMedianAbsoluteDeviation extends AbstractMultivalueFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "MvMedianAbsoluteDeviation",
+        MvMedianAbsoluteDeviation::new
+    );
+
+    @FunctionInfo(
+        returnType = { "double", "integer", "long", "unsigned_long" },
+        description = "Converts a multivalued field into a single valued field containing the median absolute deviation.",
+        note = "If the field has an even number of values, "
+            + "the medians will be calculated as the average of the middle two values. "
+            + "If the column is not floating point, the averages round towards 0.",
+        examples = @Example(file = "mv_median_absolute_deviation", tag = "example")
+    )
+    public MvMedianAbsoluteDeviation(
+        Source source,
+        @Param(
+            name = "number",
+            type = { "double", "integer", "long", "unsigned_long" },
+            description = "Multivalue expression."
+        ) Expression field
+    ) {
+        super(source, field);
+    }
+
+    private MvMedianAbsoluteDeviation(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    protected TypeResolution resolveFieldType() {
+        return isType(field(), t -> t.isNumeric() && isRepresentable(t), sourceText(), null, "numeric");
```

Comment:
I noticed that `median_absolute_deviation` doesn't accept `unsigned_long` while its `mv_` sister does. Is this a miss or has   a reason?

Reply:
`mv_median` allows unsigned_longs, but MedianAbsDev uses QuantileStates. As long as QuantileStates can work with unsigned longs, it should be possible, but I don't know